flow/connectors/utils/avro/avro_writer.go

```diff
@@ -186,14 +186,13 @@ func (p *peerDBOCFWriter) WriteOCF(w io.Writer) (int, error) {
 
 func (p *peerDBOCFWriter) WriteRecordsToS3(bucketName, key string, s3Creds utils.S3PeerCredentials) (*AvroFile, error) {
 	r, w := io.Pipe()
-	numRowsWritten := make(chan int, 1)
+	var writeOcfError error
+	var numRows int
+	var wg sync.WaitGroup
+	wg.Add(1)
 	go func() {
-		defer w.Close()
-		numRows, err := p.WriteOCF(w)
-		if err != nil {
-			log.Fatalf("%v", err)
-		}
-		numRowsWritten <- numRows
+		numRows, writeOcfError = p.WriteOCF(w)
```

Comment:
I could've had `defer wg.Done()` at top, but opted not to. The difference is that now if `WriteOCF` panics the surrounding goroutine will be stuck, rather than returning 0. The ideal would be to return an error, which would mean detecting that the goroutine ended in a panic

Another way would be to introduce a `var noPanic bool` which gets set to true at the end of the goroutine so that if after `wg.Wait()` `noPanic` is false then we can report an error

Reply:
reading over, we can rely on `w.Close` to synchronize return values